File: gabriel/Problema0437/Problema0437.java
package Problema0437;

import java.util.Scanner;
public class Problema0437 {
    public static float calcularVenda(float produtoA, float produtoB){
        float precoFinal;
        

        if(produtoA > 0  && produtoB > 0){
          precoFinal = (float) ((produtoA * 10 + produtoB * 20) * 0.85);
        }
        else{
          precoFinal = (float) ((produtoA * 10 + produtoB * 20) * 0.9);
        }
        return precoFinal;
    }

    public static void main(String[] args) {
      Scanner input = new Scanner(System.in);

    float a,b, precoFinal;
    
    System.out.println("Digite a quantidade comprada do produto A: ");
    a = Float.valueOf(input.nextLine());

    System.out.println("Digite a quantidade comprada do produto B: ");
    b = Float.valueOf(input.nextLine());

    precoFinal = calcularVenda(a, b);
    System.out.println("Preço final da compra: "+precoFinal);
  }
}

File: gabriel/Problema0437/Problema0437.py
def calcularVenda(produtoA, produtoB):
      
    if(produtoA > 0  and produtoB > 0):
          precoFinal = (produtoA * 10 + produtoB * 20) * 0.85
        
    else:
        precoFinal = (produtoA * 10 + produtoB * 20) * 0.9
        
    return precoFinal
    
   
a = int(input("Digite a quantidade comprada do produto A: "))

b = int(input("Digite a quantidade comprada do produto B: "))

precoFinal = calcularVenda(a, b)
print("Preço final da compra: "+str(precoFinal))